examples/minimal/packages/client-vanilla/src/mud/setup.ts

```diff
@@ -0,0 +1,91 @@
+import { setupMUDV2Network } from "@latticexyz/std-client";
+import { createFastTxExecutor, createFaucetService } from "@latticexyz/network";
+import { getNetworkConfig } from "./getNetworkConfig";
+import { defineContractComponents } from "./contractComponents";
+import { clientComponents } from "./clientComponents";
+import { world } from "./world";
+import { Contract, Signer, utils } from "ethers";
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { IWorld__factory } from "contracts/types/ethers-contracts/factories/IWorld__factory";
+
+export type SetupResult = Awaited<ReturnType<typeof setup>>;
+
+export async function setup() {
+  const contractComponents = defineContractComponents(world);
+  const networkConfig = await getNetworkConfig();
+  const result = await setupMUDV2Network<typeof contractComponents>({
+    networkConfig,
+    world,
+    contractComponents,
+    syncThread: "main",
+  });
+
+  result.startSync();
+
+  // Request drip from faucet
+  const signer = result.network.signer.get();
+  if (!networkConfig.devMode && networkConfig.faucetServiceUrl && signer) {
```

Comment:
I think we can remove `devMode` here and execute the if clause if a `faucetServiceUrl` is provided